src/components/EmailForm/index.tsx

```diff
@@ -20,44 +20,70 @@ type EmailData = {
   text: string;
 };
 
-export const EmailForm: React.FC<EmailFormProps> = ({ eventId }): ReactNode => {
+export const EmailForm: React.FC<EmailFormProps> = ({ eventId }) => {
   const [isLoading, setIsLoading] = useState(false);
-  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isPopupInfoOpen, setIsPopupInfoOpen] = useState(false);
   const [status, setStatus] = useState<number>();
+  const [pendingEmailData, setPendingEmailData] = useState<EmailData | null>(
+    null
+  );
+
+  const { isUserLoggedIn } = useAuth();
+  const { openAuthModal } = useAuthModal();
 
-  const formProps = useForm({
+  const formProps = useForm<EmailData>({
     resolver: yupResolver(emailSchema),
     defaultValues: {
       text: '',
     },
   });
 
-  const onSubmit = async (emailData: EmailData) => {
-    const messageDataFormatted = {
-      ...emailData,
-      eventId,
-    };
+  const submitMessage = useCallback(
+    async (emailData: EmailData) => {
+      const messageDataFormatted = {
+        ...emailData,
+        eventId,
+      };
 
-    setIsLoading(true);
+      setIsLoading(true);
 
-    try {
-      const response = await postMessage(messageDataFormatted);
+      try {
+        const response = await postMessage(messageDataFormatted);
 
-      if (response && !response.error) {
-        setStatus(200);
-      }
+        if (response && !response.error) {
+          setStatus(200);
+        }
 
-      if (response?.error) {
-        setStatus(response.error?.status);
+        if (response?.error) {
+          setStatus(response.error?.status);
+        }
+      } finally {
+        setIsPopupInfoOpen(true);
+        setIsLoading(false);
       }
-    } finally {
-      setIsLoading(false);
-      setIsModalOpen(true);
+    },
+    [eventId]
+  );
+
+  useEffect(() => {
+    if (isUserLoggedIn && pendingEmailData) {
+      submitMessage(pendingEmailData);
     }
+  }, [isUserLoggedIn, pendingEmailData, submitMessage]);
```

Comment:
Ok this is legit. This is not the same functionality as before, where you wanted the generic callback for any post-signup action, but it works. Let's roll with it

Reply:
🙌  your solution saved for the future purposes! 🙌